### PR TITLE
Ingest non-code files in code graph pipeline

### PR DESCRIPTION
**Pull request by alekszievr**
In this first version, I assume for simplicity that every non-py file is documentation.

## Summary by CodeRabbit

- **New Features**
  - Enhanced processing capabilities for both code and non-code files in the pipeline.
  - New functions added for retrieving non-Python files and user-specific datasets.

- **Bug Fixes**
  - Improved error handling for repository path existence checks.

- **Documentation**
  - Updated method signatures and added new function descriptions for clarity.

### CODE REVIEW

File: cognee/tasks/repo_processor/get_non_code_files.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    non_py_files_paths = [
+        os.path.join(root, file)
+        for root, _, files in os.walk(repo_path) for file in files if not file.endswith(".py")
+    ]
+    return non_py_files_paths
+
+
+async def get_non_code_files(data_points: list[DataPoint]):
+    for data_point in data_points:
```
Comment (borisarzentar) [inline cognee/tasks/repo_processor/get_non_code_files.py:31]: This task will run every time for every batch, which can be a bit slow on huge amount of data points. If we split the pipeline like I mentioned above, then you don't need to do this, but receive the `repo_path` as entry data into this pipeline.

Comment (alekszievr) [reply cognee/tasks/repo_processor/get_non_code_files.py]: @borisarzentar I made the suggested changes, could you please look at it again?

File: cognee/api/v1/cognify/code_graph_pipeline.py [hunk 2]
```diff
@@ -38,12 +55,37 @@ async def run_code_graph_pipeline(repo_path):
...
+    embedding_engine = get_embedding_engine()
+    cognee_config = get_cognify_config()
+    user = await get_default_user()
+
     tasks = [
         Task(get_repo_file_dependencies),
-        Task(enrich_dependency_graph, task_config={"batch_size": 50}),
+        Task(enrich_dependency_graph),
         Task(expand_dependency_graph, task_config={"batch_size": 50}),
+        Task(get_source_code_chunks, embedding_model=embedding_engine.model, task_config={"batch_size": 50}),
```
Comment (borisarzentar) [inline cognee/api/v1/cognify/code_graph_pipeline.py:66]: Where does `get_source_code_chunks` come from?

Comment (alekszievr) [reply cognee/api/v1/cognify/code_graph_pipeline.py:66]: hey @borisarzentar I messed up the commits on this PR, moved the meaningful part here: https://github.com/topoteretes/cognee/pull/395

I am closing this one now, sorry for the confusion

**Pull request closed without merging**